doc/indexing.rst

```diff
@@ -353,67 +331,129 @@ labels of the 1st and 2nd dimension, respectively. You must specify all
 dimensions or use the ellipsis in the ``loc`` specifier, e.g. in the example
 above, ``mda.loc[{'one': 'a', 'two': 0}, :]`` or ``mda.loc[('a', 0), ...]``.
 
-Multi-dimensional indexing
---------------------------
 
-xarray does not yet support efficient routines for generalized multi-dimensional
-indexing or regridding. However, we are definitely interested in adding support
-for this in the future (see :issue:`475` for the ongoing discussion).
+.. _advanced_indexing:
 
-.. _copies vs views:
+Basic and Advanced Indexing
+---------------------------
 
-Copies vs. views
-----------------
+As similar to numpy's nd-array, xarray supports two types of indexing,
```

Comment:
I'm not sure I like this framing, since in xarray I think "basic" indexing can be considered a special case of "advanced" indexing (which isn't true for NumPy). Once you make a few substitutions (e.g., `slice` -> `np.arange(*slice.indices(...))`, `unlabeled_array` -> `(dim, unlabeled_array)`, boolean dtype -> `nonzero()`), you follow only the vectorized indexing rules. Our implementation is more complicated, but users don't need to know that.